components/calendar/benches/date.rs

```diff
@@ -0,0 +1,148 @@
+
+                // Arithmetic. black_box used to avoid compiler optimization.
+                instantiated_date_buddhist.add(DateDuration::new(
+                    black_box(1),
+                    black_box(2),
+                    black_box(3),
+                    black_box(4),
+                ));
+                converted_date_buddhist.add(DateDuration::new(
+                    black_box(1),
+                    black_box(2),
+                    black_box(3),
+                    black_box(4),
+                ));
+
+                // Retrieving vals
+                let _ = instantiated_date_buddhist.year().number;
+                let _ = instantiated_date_buddhist.month().number;
+                let _ = instantiated_date_buddhist.day_of_month().0;
+                let _ = converted_date_buddhist.year().number;
+                let _ = converted_date_buddhist.month().number;
+                let _ = converted_date_buddhist.day_of_month().0;
+
+                // Conversion to ISO.
+                let _ = instantiated_date_buddhist.to_iso();
+                let _ = converted_date_buddhist.to_iso();
+            }
+        })
+    });
+
+    #[cfg(feature = "bench")]
+    group.bench_function("calendar/date/julian", |b| {
+        use icu::calendar::julian::Julian;
+
+        b.iter(|| {
+            for fx in &fxs.0 {
+                // Instantion
+                let mut instantiated_date_julian =
+                    Date::new_julian_date(fx.year, fx.month, fx.day).unwrap();
+
+                // Conversion from ISO
+                let date_iso = Date::new_iso_date_from_integers(fx.year, fx.month, fx.day).unwrap();
+                let mut converted_date_julian = Date::new_from_iso(date_iso, Julian);
+
+                // Arithmetic. black_box used to avoid compiler optimization.
+                instantiated_date_julian.add(DateDuration::new(
+                    black_box(1),
+                    black_box(2),
+                    black_box(3),
+                    black_box(4),
+                ));
+                converted_date_julian.add(DateDuration::new(
+                    black_box(1),
+                    black_box(2),
+                    black_box(3),
+                    black_box(4),
+                ));
+
+                // Retrieving vals
+                let _ = instantiated_date_julian.year().number;
+                let _ = instantiated_date_julian.month().number;
+                let _ = instantiated_date_julian.day_of_month().0;
+                let _ = converted_date_julian.year().number;
+                let _ = converted_date_julian.month().number;
+                let _ = converted_date_julian.day_of_month().0;
+
+                // Conversion to ISO.
+                let _ = instantiated_date_julian.to_iso();
+                let _ = converted_date_julian.to_iso();
+            }
+        })
+    });
+
+    // TODO: Run-through example for all calendar types.
```

Comment:
might be worth writing this generically using callbacks for date construction, if possible

Reply:
Good call -- I'm messing with a closure that looks something like:
```rust
let mut bench_date_calendar = |calendar, instantiation_fn: fn(i32, u8, u8), name| {
        group.bench_function(&format!("calendar/date/{}", name), |b| {
            b.iter(|| for fx in &fxs.0 {
                // Instantiation of {calendar} instance
                let mut instantiated_date =
                    instantiation_fn(fx.year, fx.month, fx.day).unwrap();

                // Conversion from ISO to {calendar}
                let date_iso = Date::new_iso_date_from_integers(fx.year, fx.month, fx.day).unwrap();
                let mut converted_date = Date::new_from_iso(date_iso, calendar);

                // Arithmetic. black_box used to avoid compiler optimization.
                instantiated_date.add(DateDuration::new(
                    black_box(1),
                    black_box(2),
                    black_box(3),
                    black_box(4),
                ));
                converted_date.add(DateDuration::new(
                    black_box(1),
                    black_box(2),
                    black_box(3),
                    black_box(4),
                ));

                // Retrieving vals
                let _ = instantiated_date.year().number;
                let _ = instantiated_date.month().number;
                let _ = instantiated_date.day_of_month().0;
                let _ = converted_date.year().number;
                let _ = converted_date.month().number;
                let _ = converted_date.day_of_month().0;

                // Conversion from {calendar} to ISO.
                let _ = instantiated_date.to_iso();
                let _ = converted_date.to_iso();
            })
        });
    };

    {
        bench_date_calendar(buddhist::Buddhist, Date::new_buddhist_date, "buddhist");
        bench_date_calendar(julian::Julian, Date::new_julian_date(), "julian");
    }
```

However, I'm running into issues passing the instantiation method
```
bench_date_calendar(buddhist::Buddhist, Date::new_buddhist_date, "buddhist");
                                        ^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found enum `Result`
```

Continuing to troubleshoot tonight, but finding challenges in successfully passing the function pointer

Reply:
so you won't be able to do this using a closure since closures can't be generic; it'll have to be a function which accepts `b` and `group` and everything.

The issue with the instantiation function is that your function pointer type doesn't handle the return value.

Reply:
I think the `bench_date_calendar` thing should be a function, but it should accept the constructor as a closure